### PR TITLE
Fix Clang warnings - libYARP_sig

### PR DESCRIPTION
**Pull request by damn1**
libYARP_sig - fix file by file

- after the assignment the var is never read
- possible access to null pointer
- missing initialization, the content content could be garbage

### CODE REVIEW

File: src/libYARP_sig/include/yarp/sig/ImageNetworkHeader.h [hunk 1]
```diff
@@ -46,6 +46,25 @@ class yarp::sig::ImageNetworkHeader
     yarp::os::NetInt32 paramBlobTag;
     yarp::os::NetInt32 paramBlobLen;
 
+    ImageNetworkHeader()
```
Comment (drdanz) [inline src/libYARP_sig/include/yarp/sig/ImageNetworkHeader.h]: @Nicogene Why is it complaining about this? Perhaps the actual issue is in NetInt32? Anyway, please use the initialization list instead or even better (if it fixes the issue) the defaulted constructor (`ImageNetworkHeader() = default;`)

Comment (drdanz) [reply src/libYARP_sig/include/yarp/sig/ImageNetworkHeader.h]: Oh wait, this is master... no `default`

File: src/libYARP_sig/src/Matrix.cpp [hunk 2]
```diff
@@ -36,6 +36,19 @@ class MatrixPortContentHeader
     yarp::os::NetInt32 cols;
     yarp::os::NetInt32 listTag;
     yarp::os::NetInt32 listLen;
+
+    MatrixPortContentHeader()
```
Comment (drdanz) [inline src/libYARP_sig/src/Matrix.cpp]: Same

File: src/libYARP_sig/src/Vector.cpp [hunk 3]
```diff
@@ -36,6 +36,10 @@ class VectorPortContentHeader
 public:
     yarp::os::NetInt32 listTag;
     yarp::os::NetInt32 listLen;
+    VectorPortContentHeader(){
```
Comment (drdanz) [inline src/libYARP_sig/src/Vector.cpp]: same

File: src/libYARP_sig/include/yarp/sig/ImageNetworkHeader.h (hunk 1 above)
Comment (Nicogene) [reply src/libYARP_sig/include/yarp/sig/ImageNetworkHeader.h]: Clang static analyzer complaints about the missing initialization in the constructor.

File: src/libYARP_sig/include/yarp/sig/ImageNetworkHeader.h [hunk 4]
```diff
@@ -46,6 +46,12 @@ class yarp::sig::ImageNetworkHeader
     yarp::os::NetInt32 paramBlobTag;
     yarp::os::NetInt32 paramBlobLen;
 
+    ImageNetworkHeader() : listTag(0), listLen(0), paramNameTag(0), 
+		                   paramName(0), paramIdTag(0), id(0),
```
Comment (drdanz) [inline src/libYARP_sig/include/yarp/sig/ImageNetworkHeader.h]: Please remove the 'tab' characters

File: src/libYARP_sig/src/Vector.cpp [hunk 5]
```diff
@@ -36,6 +36,7 @@ class VectorPortContentHeader
 public:
     yarp::os::NetInt32 listTag;
     yarp::os::NetInt32 listLen;
+	VectorPortContentHeader() : listTag(0), listLen(0) {}
```
Comment (drdanz) [inline src/libYARP_sig/src/Vector.cpp]: 'tab'

**Comment by drdanz** [comment]
Merged, thanks.

**Pull request merged**